crates/ruff_python_semantic/src/model.rs

```diff
@@ -257,6 +257,27 @@ impl<'a> SemanticModel<'a> {
             .is_some_and(|binding| binding.kind.is_builtin())
     }
 
+    /// Return `true` if `member` is a reference to `builtins.$target`,
+    /// i.e. either `object` (where `object` is not overridden in the global scope),
+    /// or `builtins.object` (where `builtins` is imported as a module at the top level)
+    pub fn match_builtin_expr(&self, expr: &Expr, symbol: &str) -> bool {
```

Comment:
I wonder if we could instead change the method to return a `QualifiedName` where constructing the `QualifiedName` short-circuits if `Modules::BUILTINS` isn't imported. 

I'm bringing this up because `QualifiedName` already has methods for handling builtins (that might be incorrect?)

```
pub fn is_builtin(&self) -> bool {
    matches!(self.segments(), ["", ..])
}
```

So what I would have in mind is

```
model.resolve_builtin_name(expr).is_some_and(|builtin| builtin.is_builtin_name("open"))
```

We could still have a helper as you have today to avoid some of the repetition (`is_some_and` is somewhat annoying). But it would build up on the same concepts.

Reply:
I like it.